internal/buildpack/sbt.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2021 YourBase Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package buildpack
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"os"
+	"io/ioutil"
+
+	"github.com/yourbase/yb"
+	"github.com/yourbase/yb/internal/biome"
+	"zombiezen.com/go/log"
+)
+
+func installSBT(ctx context.Context, sys Sys, spec yb.BuildpackSpec) (_ biome.Environment, err error) {
+	installDir := sys.Biome.JoinPath(sys.Biome.Dirs().Tools, "sbt", spec.Version())
+	desc := sys.Biome.Describe()
+	parentDir := os.TempDir()
+	socketDir, err := ioutil.TempDir(parentDir, "*-ybsbt")
+	if err != nil {
+		log.Errorf(ctx, "Error creating directory for SBT sockets: %v", err)
+	}
+	log.Infof(ctx, "SBT will use %s as the socket directory", socketDir)
```

Comment:
I don't think the average user needs to know this information unless something went wrong, so I suggest debug-level log.

```suggestion
	log.Debugf(ctx, "SBT will use %s as the socket directory", socketDir)
```